File: 프로그래머스/1/17681. ［1차］ 비밀지도/［1차］ 비밀지도.java
class Solution {
    public String[] solution(int n, int[] arr1, int[] arr2) {
        String[] answer = new String[n];
        for(int i=0; i<n; i++){
            answer[i] = "";
        }
        //입력값 두개를 전부 이진수로 변환
        for(int i=0; i<n; i++){
            for(int j=0; j<n; j++){
                if(arr1[i] % 2 == 1 || arr2[i] % 2 == 1) {
                    answer[i] = "#" + answer[i];
                } //둘중 하나라도 1이면 색칠
                else{
                    answer[i] = " " + answer[i];
                }
                arr1[i] /= 2;
                arr2[i] /= 2;
            }
        }

        return answer;
    }
}

File: 프로그래머스/1/17681. ［1차］ 비밀지도/［1차］ 비밀지도.cpp
#include <string>
#include <vector>
#include<iostream>
#include <bits/stdc++.h>
using namespace std;

vector<string> solution(int n, vector<int> arr1, vector<int> arr2) {
    vector<string> answer(n,"");
    for(int i=0; i<n; i++){
        int cmp = pow(2,n-1);
        for(int j=0; j<n; j++){
            if(arr1[i] < cmp && arr2[i] < cmp){
            answer[i] += " ";
            }
            else{
                if(arr1[i] >= cmp){
                    arr1[i] -= cmp;
                }
                if(arr2[i] >= cmp){
                    arr2[i] -= cmp;
                }
                answer[i] += "#";
            }
        cmp /= 2;
        }
    }
    return answer;
}